src/dcd/server/autocomplete/util.d

```diff
@@ -143,8 +144,14 @@ SymbolStuff getSymbolsForCompletion(const AutocompleteRequest request,
 	ScopeSymbolPair pair = generateAutocompleteTrees(tokenArray,
 		rba, request.cursorPosition, moduleCache);
 	auto expression = getExpression(beforeTokens);
-	return SymbolStuff(getSymbolsByTokenChain(pair.scope_, expression,
-		request.cursorPosition, type), pair.symbol, pair.scope_);
+	auto symbols = getSymbolsByTokenChain(pair.scope_, expression,
+		request.cursorPosition, type);
+		if (symbols.length == 0 && doUFCSSearch(stringToken(beforeTokens.front), stringToken(beforeTokens.back))) {
+			// Let search for UFCS, since we got no hit
+			symbols ~= getSymbolsByTokenChain(pair.scope_, getExpression([beforeTokens.back]),
```

Comment:
@WebFreak001 

This is only useful in case of ambiguity with UFCS function and method, where we run into the case of more than one goto definition. 
example here we are having the function and method fooHello which has the same name and parameters:
```
module barutils;
struct Foo
{
	void fooHello() {}
} 
void fooHello(Foo foo) {}
```

```
import barutils;

void main()
{
	auto foo = Foo();
	foo.fooHello;
}
```

goto definition on foo.fooHello;

right now with the current implementation it will only goto definition of the one within the struct.

I made this solution since the compiler will also choose the one within the struct if there is ambiguity with method and UFCS function.

we also  require changes for the client and messages.d since the client only supports for one goto definition.

Inside message.d : 165, we need to have arrays of symbolFilePaths and symbolLcations

```
	/**
	 * The path to the file that contains the symbol.
	 */
	string symbolFilePath;

	/**
	 * The byte offset at which the symbol is located.
	 */
	size_t symbolLocation;

```
does it clarify it?

I do have an idea about how to implement more goto definitions, later on if needed.